libafl/src/bolts/launcher.rs

```diff
@@ -112,8 +112,10 @@ where
         println!("spawning on cores: {:?}", self.cores);
 
         // Spawn clients
-        for (index, (id, bind_to)) in core_ids.iter().enumerate().take(num_cores).enumerate() {
```

Comment:
Why not keep `enumerate`?
I don't care much but manually enumerating seems less "rusty"

Reply:
`enumerate` doesn't do what you expect in this case. The manual enumeration is the simplest solution I could find....

Reply:
The reason `enumerate` doesn't do what you expect is because we filter AFTER this point, see the next line.

Reply:
Instead of manually filtering with an `if`, we could just add the filter to the iterator cahin

Reply:
yeah. I'm too tired to do that...